lib/rules/amd-function-arity.js

```diff
@@ -30,10 +31,41 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
+        if (allowExtraDependencies) {
+            return false;
+        }
+
+        if (context.options && context.options[0] && context.options[0].allowExtraDependencies) {
+            return context.options[0].allowExtraDependencies;
+        }
+
+        return false;
+    }
+
     function requireHasCallback(node) {
         return node.arguments.filter(isFunctionExpression).length;
     }
 
+    function getExtraPaths(dependencyNodes, callbackParams) {
```

Comment:
It seems like you're trying to get the tail of the array here. But rather than iterate to build up a new array, would `Array.prototype.slice` work here? 

For example:
```js
function getUnassignedPaths(paths, args) {
    return paths.slice(args.length - paths.length);
}
```